tests/numeric/test_PCATransformer.py

```diff
@@ -0,0 +1,416 @@
+        ):
+            PCATransformer(columns="b", random_state="2")
+
+    def test_to_svd_solver_type_error(self):
+        """Test that an exception is raised if svd_solver is not a str."""
+
+        with pytest.raises(
+            TypeError,
+            match=r"""PCATransformer:unexpected type <class 'int'> for svd_solver, must be str""",
+        ):
+            PCATransformer(columns="b", svd_solver=2)
+
+    def test_to_n_components_type_error(self):
+        """Test that an exception is raised if n_components is not a int or 'mle'."""
+
+        with pytest.raises(
+            TypeError,
+            match=r"""PCATransformer:unexpected type <class 'str'> for n_components, must be int or equal to 'mle'.""",
+        ):
+            PCATransformer(columns="b", n_components="3")
+
+    def test_to_svd_solver_value_error(self):
+        """Test that an exception is raised if svd_solver is not one of the allowed values."""
+
+        with pytest.raises(
+            ValueError,
+            match=r"""PCATransformer:svd_solver zzz is unknown. Please select among 'auto', 'full', 'arpack', 'randomized'.""",
+        ):
+            PCATransformer(columns="b", svd_solver="zzz")
+
+    def test_to_n_components_value_error(self):
+        """Test that an exception is raised if n_components is not one of the allowed values."""
+
+        with pytest.raises(
+            ValueError,
+            match=r"""PCATransformer:n_components must be positive  got -1""",
+        ):
+            PCATransformer(columns="b", n_components=-1)
+
+    def test_super_init_called(self, mocker):
+        """Test that super.__init__ called."""
+
+        expected_call_args = {
+            0: {
+                "args": (),
+                "kwargs": {"columns": ["a", "b"], "copy": True, "verbose": False},
+            }
+        }
+
+        with ta.functions.assert_function_call(
+            mocker, tubular.base.BaseTransformer, "__init__", expected_call_args
+        ):
+            PCATransformer(columns=["a", "b"], n_components=1, copy=True, verbose=False)
+
+
+class TestCheckNumericColumns(object):
+    """Tests for the check_numeric_columns method."""
+
+    def test_arguments(self):
+        """Test that check_numeric_columns has expected arguments."""
+
+        ta.functions.test_function_arguments(
+            func=PCATransformer.check_numeric_columns,
+            expected_arguments=["self", "X"],
+            expected_default_values=None,
+        )
+
+    def test_exception_raised(self):
+        """Test an exception is raised if non numeric columns are passed in X."""
+
+        df = d.create_df_2()
+
+        x = PCATransformer(columns=["a", "b", "c"], n_components=2)
+
+        with pytest.raises(
+            TypeError,
+            match=r"""PCATransformer: The following columns are not numeric in X; \['b', 'c'\]""",
+        ):
+            x.check_numeric_columns(df)
+
+    def test_X_returned(self):
```

Comment:
As in comment on the call to this method in fit.  I'm not sure we need to return X or do assignment.

Reply:
related to a new git issue check_numeric_columns ?